.github/workflows/create-wth-template.sh

```diff
@@ -234,27 +247,30 @@ while getopts ":c:dhn:p:v" option; do
     d) deleteExistingDirectoryArg=true;;
     h) Help
        exit;;
-    n) nameOfChallengeArg=${OPTARG};;
+    n) nameOfHackArg=${OPTARG};;
     p) pathArg=${OPTARG};;
     v) verbosityArg=true
   esac
 done
 
 if $verbosityArg; then
   echo "Number of Challenges: $numberOfChallengesArg"
-  echo "Name of Challenge: $nameOfChallengeArg"
+  echo "Name of Challenge: $nameOfHackArg"
   echo "Path: $pathArg"
   echo "Delete existing directory: $deleteExistingDirectoryArg"
 fi
 
-declare -r wthDirectoryName="xxx-$nameOfChallengeArg"
+declare -r wthDirectoryName="xxx-$nameOfHackArg"
 
 declare -r rootPath="$pathArg/$wthDirectoryName"
 
 declare -r pathToTemplateDirectory="$pathArg/$templateDirectoryName"
 
 CreateDirectoryStructure $deleteExistingDirectoryArg
 
-CreateHackDescription $numberOfChallengesArg
+# + 1 to account for the including the prerequisites file
+declare -r totalNumberOfChallenges=$(($numberOfChallengesArg + 1))
+
+CreateHackDescription $totalNumberOfChallenges
 
-CreateChallengesAndSolutions $numberOfChallengesArg
+CreateChallengesAndSolutions $totalNumberOfChallenges
```

Comment:
Not sure how/where you're setting it, but current version is outputting n+1 challenges.   I requested 10 challenges, I got 11 plus challenge zero, making a total of 12.  I'm assuming one of the counters is off because of the new challenge Zero?

Reply:
done